tests/test_adjoint.py

```diff
@@ -71,28 +71,23 @@ def test_adjoint(problem, method, sde_type, noise_type, adaptive):
         'log_ode': 'foster'
     }[method]
     bm = torchsde.BrownianInterval(
-        t0=t0, t1=t1, shape=(batch_size, d), dtype=dtype, device=device,
+        t0=t0, t1=t1, shape=(batch_size, m), dtype=dtype, device=device,
         levy_area_approximation=levy_area_approximation
     )
-    with torch.no_grad():
-        grad_outputs = torch.ones(batch_size, d).to(device)
-        alt_grad = problem.analytical_grad(y0, t1, grad_outputs, bm)
 
-    problem.zero_grad()
-    _, yt = torchsde.sdeint_adjoint(problem, y0, ts, bm=bm, method=method, dt=dt, adaptive=adaptive)
-    loss = yt.sum(dim=1).mean(dim=0)
-    loss.backward()
-    adj_grad = torch.cat(tuple(p.grad for p in problem.parameters()))
-    assert_allclose(alt_grad, adj_grad)
+    def func(inputs, modules):
+        y0, sde = inputs[0], modules[0]
+        ys = torchsde.sdeint_adjoint(sde, y0, ts, bm, dt=dt, method=method, adaptive=adaptive)
+        return (ys[-1] ** 2).sum(dim=1).mean(dim=0)
+
+    # `grad_inputs=True` also works, but we really only care about grad wrt params and want fast tests.
+    utils.gradcheck(func, y0, sde, eps=1e-6, rtol=1e-2, atol=1e-3, grad_params=True)
 
 
 @pytest.mark.parametrize("problem", [BasicSDE1, BasicSDE2, BasicSDE3, BasicSDE4])
 @pytest.mark.parametrize("method", ito_methods.keys())
 @pytest.mark.parametrize('adaptive', (False, True))
```

Comment:
Should adaptive=True be removed here? (Given that the Euler check is removed below.)

Reply:
`adaptive=True` shouldn't affect that test. I removed it for the test above since it affects the finite-difference gradients. The above tests pass if I use the analytical solutions. But not every problem has an analytical solution, and maintaining that code might become tricky.